integrationTests/chainSimulator/staking/stakeAndUnStake_test.go

```diff
@@ -0,0 +1,279 @@
+		ApiInterface:           api.NewNoApiInterface(),
+		MinNodesPerShard:       100,
+		MetaChainMinNodes:      100,
+		AlterConfigsFunction: func(cfg *config.Configs) {
+			cfg.SystemSCConfig.StakingSystemSCConfig.NodeLimitPercentage = 1
+			cfg.GeneralConfig.ValidatorStatistics.CacheRefreshIntervalInSec = 1
+		},
+	})
+	require.Nil(t, err)
+	require.NotNil(t, cm)
+
+	err = cm.GenerateBlocks(150)
+	require.Nil(t, err)
+
+	// Step 1 --- add a new validator key in the chain simulator
+	numOfNodes := 20
+	validatorSecretKeysBytes, blsKeys := helpers.GenerateBlsPrivateKeys(t, numOfNodes)
+	err = cm.AddValidatorKeys(validatorSecretKeysBytes)
+	require.Nil(t, err)
+
+	newValidatorOwner := "erd1l6xt0rqlyzw56a3k8xwwshq2dcjwy3q9cppucvqsmdyw8r98dz3sae0kxl"
+	newValidatorOwnerBytes, _ := cm.GetNodeHandler(0).GetCoreComponents().AddressPubKeyConverter().Decode(newValidatorOwner)
+	rcv := "erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqplllst77y4l"
+	rcvAddrBytes, _ := cm.GetNodeHandler(0).GetCoreComponents().AddressPubKeyConverter().Decode(rcv)
+
+	// Step 2 --- set an initial balance for the address that will initialize all the transactions
+	err = cm.SetStateMultiple([]*dtos.AddressState{
+		{
+			Address: "erd1l6xt0rqlyzw56a3k8xwwshq2dcjwy3q9cppucvqsmdyw8r98dz3sae0kxl",
+			Balance: "1000000000000000000000000",
+		},
+	})
+	require.Nil(t, err)
+
+	// Step 3 --- generate and send a stake transaction with the BLS keys of the validators key that were added at step 1
+	validatorData := ""
+	for _, blsKey := range blsKeys {
+		validatorData += fmt.Sprintf("@%s@010101", blsKey)
+	}
+
+	numOfNodesHex := hex.EncodeToString(big.NewInt(int64(numOfNodes)).Bytes())
+	stakeValue, _ := big.NewInt(0).SetString("51000000000000000000000", 10)
+	tx := &transaction.Transaction{
+		Nonce:     0,
+		Value:     stakeValue,
+		SndAddr:   newValidatorOwnerBytes,
+		RcvAddr:   rcvAddrBytes,
+		Data:      []byte(fmt.Sprintf("stake@%s%s", numOfNodesHex, validatorData)),
+		GasLimit:  500_000_000,
+		GasPrice:  1000000000,
+		Signature: []byte("dummy"),
+		ChainID:   []byte(configs.ChainID),
+		Version:   1,
+	}
+
+	txFromNetwork := helpers.SendTxAndGenerateBlockTilTxIsExecuted(t, cm, tx, maxNumOfBlockToGenerateWhenExecutingTx)
+	require.NotNil(t, txFromNetwork)
+
+	err = cm.GenerateBlocks(1)
+	require.Nil(t, err)
+
+	results, err := cm.GetNodeHandler(core.MetachainShardId).GetFacadeHandler().AuctionListApi()
+	require.Nil(t, err)
+	require.Equal(t, newValidatorOwner, results[0].Owner)
+	require.Equal(t, 20, len(results[0].AuctionList))
+	totalQualified := 0
+	for _, res := range results {
+		for _, node := range res.AuctionList {
+			if node.Qualified {
+				totalQualified++
+			}
+		}
+	}
+	require.Equal(t, 8, totalQualified)
+
+	err = cm.GenerateBlocks(100)
+	require.Nil(t, err)
+
+	results, err = cm.GetNodeHandler(core.MetachainShardId).GetFacadeHandler().AuctionListApi()
+	require.Nil(t, err)
+
+	totalQualified = 0
+	for _, res := range results {
```

Comment:
for in for. Can we extract this in a function? And reused on L255

Reply:
done